### PR TITLE
feat(sms): add migration task resource

### PR DESCRIPTION
**Pull request by ShiChangkuo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ export HW_SMS_SOURCE_SERVER="ecs-sms-test"
$ make testacc TEST='./huaweicloud/services/acceptance/sms' TESTARGS='-run TestAccMigrationTask_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/sms -v -run TestAccMigrationTask_basic -timeout 360m -parallel 4
=== RUN   TestAccMigrationTask_basic
=== PAUSE TestAccMigrationTask_basic
=== CONT  TestAccMigrationTask_basic
--- PASS: TestAccMigrationTask_basic (31.16s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/sms     31.239s

```

### CODE REVIEW

File: docs/resources/sms_task.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+* `enterprise_project_id` - The enterprise project id of the target server.
+* `target_server_name` - The name of the target server.
+* `migrate_speed` - The migration rate, in MB/s.
+
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `create` - Default is 5 minute.
+* `update` - Default is 5 minute.
```
Comment (niuzhenguo) [inline docs/resources/sms_task.md]: seems update timeout does not take effect like create

File: docs/resources/sms_task.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  The [object](#physical_volumes_object) is documented below. Changing this parameter will create a new resource.
+
+<a name="physical_volumes_object"></a>
+The `physical_volumes` block supports:
+
+* `name` - (Required, String, ForceNew) Specifies the volume name. In Windows, it indicates the drive letter,
+  and in Linux, it indicates the device ID, e.g. "/dev/sda1".
+  Changing this parameter will create a new resource.
+
+* `size` - (Required, int, ForceNew) Specifies the volume size in bytes. Changing this parameter will create a new resource.
```
Comment (niuzhenguo) [inline docs/resources/sms_task.md]: Not sure what unit we use on console side, but bytes for disk seems a bit inconvinient

File: docs/resources/sms_task.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+* `target_server_disks` - (Optional, List, ForceNew) Specifies the disk configurations of the target server.
+  If omitted, it will be obtained from the source server. The [object](#target_server_disks_object)
+  is documented below. Changing this parameter will create a new resource.
+
+* `use_public_ip` - (Optional, Bool, ForceNew) Specifies whether to use a public IP address for migration.
+  The default value is `true`. Changing this parameter will create a new resource.
+
+* `migration_ip` - (Optional, String, ForceNew) Specifies the IP address of the target server.
+  Use the EIP of the target server if the migration network type is Internet.
+  Use the private IP address the target server if the migration network type is Direct Connect or VPN.
```
Comment (niuzhenguo) [inline docs/resources/sms_task.md]: Nits: missing of between address and the target server. The private IP address of the targer server

File: docs/resources/sms_task.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+  Changing this parameter will create a new resource.
+
+* `start_target_server` - (Optional, Bool, ForceNew) Specifies whether to start the target server after the migration.
+  The default value is `true`. Changing this parameter will create a new resource.
+
+* `syncing` - (Optional, Bool, ForceNew) - Specifies whether to perform a continuous synchronization after the first replication.
+  The default value `false`. Changing this parameter will create a new resource.
```
Comment (niuzhenguo) [inline docs/resources/sms_task.md]: Nits: s/The default value false/The default value is false

File: docs/resources/sms_task.md (hunk 1 above)
Comment (ShiChangkuo) [reply docs/resources/sms_task.md]: fixed

File: docs/resources/sms_task.md (hunk 2 above)
Comment (ShiChangkuo) [reply docs/resources/sms_task.md]: convert the unit of disk size toMB

File: docs/resources/sms_task.md (hunk 3 above)
Comment (ShiChangkuo) [reply docs/resources/sms_task.md]: fixed

File: docs/resources/sms_task.md (hunk 4 above)
Comment (ShiChangkuo) [reply docs/resources/sms_task.md]: fixed

**Pull request merged**